contrib/systemd/etcd2-backup-coreos/README.md

```diff
@@ -0,0 +1,255 @@
+# etcd2-backup-coreos
+
+Remote backup and multi-node restore services for etcd2 clusters on CoreOS Linux.
+
+**Warning:** This package is only intended for installation on CoreOS Linux.
+
+## Terminology
+
+**founding member** : The node which is the first member of the new recovered cluster. It is this node's rclone backup data (only) that will be used to restore the cluster. The rest of the nodes will join the cluster with no data, and simply catch up with the **founding member**.
+
+## Configuration
+Before installing etcd2-backup, you need to configure `30-etcd2-backup-restore.conf`.
+
+```
+[Service]
+Environment="ETCD_RESTORE_MASTER_ADV_PEER_URLS=<http://host:port>"
+Environment="RCLONE_ENDPOINT=remote-name:path/to/backups"
+```
+
+Assuming you're deploying to CoreOS with etcd2, you should only need to change
+
+- `ETCD_RESTORE_MASTER_ADV_PEER_URLS`
+   This is the new advertised peer url of the new etcd2 node that will be the founding member of the new restored cluster. We will call this node the **founding member**.
+
+-  `RCLONE_ENDPOINT`
+    The rclone endpoint to which backups will be stored.
+
+    Feel free to point any number of machines as the same RCLONE_ENDPOINT, path and all. Backups for each machine are stored in a sub-folder named with the machine ID (%m in systemd parlance)
+
+-  `./rclone.conf`
+    The rclone configuration file which will be installed. Must list a `[section]` which matches `RCLONE_ENDPOINT`'s remote-name component.
+
+    An easy way to generate this config file is to [install rclone](http://rclone.org/install/) on your local machine. Then follow the [configuration instructions](http://rclone.org/docs/) to generate an `rclone.conf` file.
+
+If you want to adjust backup frequency, edit `./etcd2-backup.timer`
+
+## Installation
+Once you've got those things configured, you can run `./build`.
+
+That script will generate a tarball can be copied to CoreOS instances and contains a script `etcd2-backup-install`
+
+After the tarball is inflated and the install script exected, you should see three new systemd services - one used for periodically backing up and two for restoring the cluster.
```

Comment:
After extracting the contents of the tar file and running the install script, three new systemd services are added. One service preforms periodic etcd backups, while the other two services handle restore procedures.